### PR TITLE
Add 'kadena address'

### PR DESCRIPTION
**Pull request by mankyKitty**
Adding the types and underlying functionality for a 'kadena account'.

### CODE REVIEW

File: frontend/src/Frontend/KadenaAddress.hs [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+parseKadenaAddressBlob :: A.Parser (AccountName, ChainId, NetworkName, Word32)
+parseKadenaAddressBlob = (,,,)
+  <$> parseAccountName
+  <*> parseChainId
+  <*> parseNetworkName
+  <*> A.hexadecimal
+  where
+    failParse = either fail pure
+    mkParser p c = (c <$> p) >>= failParse
```
Comment (tomsmalley) [inline frontend/src/Frontend/KadenaAddress.hs]: I needed to add a type signature here to get it to compile

File: frontend/src/Frontend/KadenaAddress.hs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  , _kadenaAddress_checksum :: Word32
+    -- ^ The CRC32 checksum of the AccountName, ChainId, and NetworkName
+  }
+  deriving (Show, Eq)
+
+textKadenaAddress :: KadenaAddress -> Text
+textKadenaAddress = TE.decodeUtf8With TE.lenientDecode . _kadenaAddress_encoded
+
+humanReadableDelimiter :: Word8
+humanReadableDelimiter = 93 -- ']'
```
Comment (tomsmalley) [inline frontend/src/Frontend/KadenaAddress.hs:123]: I'd have a minor preference for `|`. No reason other than it looks better :smile:

**Review by tomsmalley — DISMISSED** [review]
Looks good. We should also restrict AccountName to latin1.

File: frontend/src/Frontend/KadenaAddress.hs (hunk 2 above)
Comment (tomsmalley) [reply frontend/src/Frontend/KadenaAddress.hs:123]: Come to think of it, account names can have this character.

**Review by tomsmalley — CHANGES_REQUESTED** [review]
We can't have the account name encoded as latin1 and also use a latin1 delimiter without doing some form of escaping.

File: frontend/src/Frontend/KadenaAddress.hs (hunk 2 above)
Comment (mankyKitty) [reply frontend/src/Frontend/KadenaAddress.hs:123]: Account names can currently have everything in latin1, so this was a placeholder at best. :< Also I agree that `|` looks better. I'll make a ticket.

**Comment by tomsmalley** [comment]
Merging since we have tickets to address the delimiter concerns

**Pull request merged**